### PR TITLE
[codex] Add searchable model picker and cap OpenRouter models

### PR DESCRIPTION
**Pull request by lsm**
Caps API-loaded OpenRouter models to 30 entries from known high-signal provider families, and adds text search to the status-bar and fallback-settings model pickers. The workflow visual editor remains on a native select with a code note because the OpenRouter list is now capped server-side.

Validation: `bun run check`; `bun run --filter @neokai/web test`; `bun run --filter @neokai/daemon test:unit`; targeted web picker tests and OpenRouter provider tests. CI is green on PR #1715.

### CODE REVIEW

**Review by lsm — COMMENTED** [review]
## 🤖 Review by Claude Opus 4.7 (Anthropic)

> **Model:** Claude Opus 4.7 | **Client:** NeoKai | **Provider:** Anthropic

**Verdict: REQUEST_CHANGES**

The implementation is well-structured overall — the search filter logic is clean, the curation approach is sound, and the tests cover the core paths. However, there is a layout regression that will break scrolling for longer model lists, plus a couple of secondary items to address.

---

### P1 — Flex layout regression breaks dropdown scrolling

**`packages/web/src/components/SessionStatusBar.tsx:485`**

The original dropdown had `overflow-y-auto` directly on the outer container (`max-h-[60vh] overflow-y-auto`), which scrolled correctly. This PR refactored the container into a `flex flex-col` layout with the search input as a sibling of the model list, but the inner scrollable div (line 485) is missing `flex-1 min-h-0`:

```tsx
// Current (broken):
<div class="overflow-y-auto">

// Fix:
<div class="flex-1 min-h-0 overflow-y-auto">
```

In a CSS flex column, items default to `min-height: auto`, which prevents them from shrinking below their content size. Without `flex-1 min-h-0`, the inner div grows to its natural content height and overflows the `max-h-[60vh]` parent instead of scrolling within it. The model list will visually spill past the dropdown boundary with no scrollbar.

### P2 — `CURATED_PROVIDER_PREFIXES` missing notable providers

**`packages/daemon/src/lib/providers/openrouter-provider.ts:34-41`**

The task spec says "models from known families: anthropic, openai, google, deepseek, meta, mistral" — and the implementation matches that list. However, OpenRouter hosts several other high-signal providers that users would expect to see:

- `xai/` — Grok models (very popular on OpenRouter)
- `cohere/` — Command R models
- `qwen/` — Qwen models (major LLM family)

The fallback behavior (raw models capped at 30) still prevents UI overload, so this isn't blocking — but adding `xai/` at minimum would be worthwhile given Grok's popularity.

### P2 — `useFilteredModelsForPicker` hook has no test coverage

**`packages/web/src/hooks/__tests__/useModelSwitcher.test.ts`**

The hook is the primary composition point in `SessionStatusBar.tsx` that chains auth filtering → search filtering. Both underlying functions (`filterModelsForPicker`, `filterModelsBySearch`) are individually well-tested, but the composition itself is untested. A test verifying that auth-filtered results are correctly further-filtered by search terms would close this gap.

---

### What looks good

- **Search filter logic**: `filterModelsBySearch` handles edge cases well (whitespace-only, null fields via `.filter(Boolean)`, multi-term AND logic). Clean implementation.
- **`useMemo` dependencies**: `useFilteredModelsForPicker` correctly lists all reactive deps. The `Map` reference only changes on RPC fetch, so no spurious re-renders.
- **Search reset**: Three reset paths (dropdown close, model switch, open) correctly clear the query.
- **`FallbackModelsSettings.tsx`**: The `hasUnselectedModels` logic correctly distinguishes "no search matches" from "all models already selected".
- **Empty state**: "No matching models" shown when `filteredModels.length === 0` — correct.
- **OpenRouter tests**: Both curation paths (curated match and fallback) are well-tested with correct length and content assertions.
- **WorkflowModelSelect comment**: Appropriate documentation of the intentional native-select limitation.
- **Test results**: Web tests all pass (207 files, 5880 tests). The single daemon test failure (`codex-anthropic-bridge/server.test.ts`) is pre-existing on `dev` and unrelated to this PR.

File: packages/web/src/components/SessionStatusBar.tsx [hunk 1]
```diff
@@ -453,74 +469,85 @@ export default function SessionStatusBar({
...
+									<input
+										type="search"
+										value={modelSearchQuery}
+										onInput={(e) => setModelSearchQuery(e.currentTarget.value)}
+										placeholder="Search models..."
+										aria-label="Search models"
+										class="w-full bg-dark-900 border border-dark-600 rounded px-2 py-1.5 text-xs text-gray-100 placeholder:text-gray-500 focus:outline-none focus:border-blue-500"
+									/>
+								</div>
+								<div class="overflow-y-auto">
```
Comment (lsm) [inline packages/web/src/components/SessionStatusBar.tsx]: **P1 — Missing `flex-1 min-h-0`** — This inner scrollable div needs both `flex-1` (to fill remaining space in the flex column) and `min-h-0` (to override the default `min-height: auto` that prevents flex items from shrinking below content size). Without these, when the model list exceeds the remaining space within `max-h-[60vh]`, the div grows to its natural height and overflows the parent instead of scrolling.\n\n```diff\n- <div class="overflow-y-auto">\n+ <div class="flex-1 min-h-0 overflow-y-auto">\n```\n\nThis is a regression from the original code, which had `overflow-y-auto` directly on the outer container.

File: packages/daemon/src/lib/providers/openrouter-provider.ts [hunk 2]
```diff
@@ -31,6 +31,15 @@ interface OpenRouterModelsResponse {
 	data?: OpenRouterModel[];
 }
 
+const CURATED_PROVIDER_PREFIXES = [
```
Comment (lsm) [inline packages/daemon/src/lib/providers/openrouter-provider.ts:34]: **P2 — Consider adding `xai/`** — Grok models are very popular on OpenRouter. The fallback cap still prevents UI overload, but users may expect to see Grok in the curated list. `cohere/` and `qwen/` are also worth considering.

File: packages/web/src/hooks/__tests__/useModelSwitcher.test.ts [hunk 3]
```diff
@@ -1224,3 +1225,42 @@ describe('filterModelsForPicker', () => {
 		expect(result).toHaveLength(2);
```
Comment (lsm) [inline packages/web/src/hooks/__tests__/useModelSwitcher.test.ts:1226]: **P2 — Missing test for `useFilteredModelsForPicker`** — This hook is the primary composition point used by `SessionStatusBar.tsx` (auth filter → search filter). Both underlying functions are well-tested individually, but the composition is untested. A test verifying that auth-filtered results are correctly further-filtered by search would close this gap.

File: packages/web/src/components/SessionStatusBar.tsx (hunk 1 above)
Comment (lsm) [reply packages/web/src/components/SessionStatusBar.tsx]: Fixed in 65fa108f3: the model dropdown scroll container now uses flex-1 min-h-0 overflow-y-auto so it is constrained by the max-height flex parent and scrolls instead of expanding.

File: packages/daemon/src/lib/providers/openrouter-provider.ts (hunk 2 above)
Comment (lsm) [reply packages/daemon/src/lib/providers/openrouter-provider.ts:34]: Fixed in 65fa108f3: added xai/, cohere/, and qwen/ to the curated OpenRouter provider prefixes.

File: packages/web/src/hooks/__tests__/useModelSwitcher.test.ts (hunk 3 above)
Comment (lsm) [reply packages/web/src/hooks/__tests__/useModelSwitcher.test.ts:1226]: Fixed in 65fa108f3: added a renderHook test for useFilteredModelsForPicker that verifies auth filtering is applied before search filtering.

File: packages/daemon/src/lib/providers/openrouter-provider.ts [hunk 4]
```diff
@@ -267,4 +279,14 @@ export class OpenRouterProvider implements Provider {
 			available: true,
 		};
 	}
+
+	private static curateApiModels(models: ModelInfo[]): ModelInfo[] {
+		const curated = models.filter((model) => {
+			const id = model.id.toLowerCase();
+			return CURATED_PROVIDER_PREFIXES.some((prefix) => id.startsWith(prefix));
+		});
```
[the remark this answers is not in the document]
Comment (lsm) [reply packages/daemon/src/lib/providers/openrouter-provider.ts:295]: Fixed in 65fa108f3: OpenRouter curation now explicitly keeps openrouter/auto on the API success path, with test coverage alongside the added popular provider families.

**Review by lsm — COMMENTED** [review]
## 🤖 Review by Claude Opus 4.7 (Anthropic)

> **Model:** Claude Opus 4.7 | **Client:** NeoKai | **Provider:** Anthropic

**Verdict: APPROVE** ✅

All three findings from round 1 are addressed:

- **P1 fixed**: `SessionStatusBar.tsx:485` — inner scroll div now has `flex-1 min-h-0 overflow-y-auto`, restoring proper scroll behavior within the flex column layout.
- **P2 fixed**: `openrouter-provider.ts:34-46` — `xai/`, `cohere/`, `qwen/` added to `CURATED_PROVIDER_PREFIXES`, plus `CURATED_MODEL_IDS` for `openrouter/auto`. New test verifies the expanded curation.
- **P2 fixed**: `useModelSwitcher.test.ts:1269-1300` — `useFilteredModelsForPicker` now tested with auth-filtered + search-filtered composition.

Tests: web 5881 pass, daemon 11613 pass (0 fail). CI green. Ready to ship.
